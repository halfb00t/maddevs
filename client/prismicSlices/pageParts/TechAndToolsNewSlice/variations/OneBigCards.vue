<template>
  <div class="one-big-cards">
    <BigCard
      v-for="(card, i) in bigCards.slice(0, 1)"
      :key="card.title + i"
      v-bind="card"
      :one-cards="true"
    />
    <LittleCard
      v-for="(card, i) in littleCards"
      :key="card.title + i"
      :title="card.title"
      :icon="card.icon"
    />
  </div>
</template>
<script>
import BigCard from '@/prismicSlices/pageParts/TechAndToolsNewSlice/components/BigCard'
import LittleCard from '@/prismicSlices/pageParts/TechAndToolsNewSlice/components/LittleCard'

export default {
  name: 'OneBigCards',

  components: {
    BigCard,
    LittleCard,
  },

  props: {
    bigCards: {
      type: Array,
      default: () => [],
    },

    littleCards: {
      type: Array,
      default: () => [],
    },
  },
}
</script>
<style lang="scss" scoped>
.one-big-cards {
  margin-bottom: 20px;
  @include grid(repeat(10, 1fr), auto, 20px, 20px);
  @media screen and (max-width: 1170px) {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    row-gap: 20px;
    column-gap: 15px;
  }

  @media screen and (max-width: 1214px) {
    ::v-deep .technologies__little-card {
      min-width: auto;
    }
  }

  @media screen and (min-width: 1170px) {
    ::v-deep .technologies__little-card-icon {
      min-width: 20px;
    }
  }
}
</style>
