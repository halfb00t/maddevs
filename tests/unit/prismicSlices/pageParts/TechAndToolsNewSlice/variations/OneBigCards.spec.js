import { render, screen } from '@testing-library/vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import OneBigCards from '@/prismicSlices/pageParts/TechAndToolsNewSlice/variations/OneBigCards'
// eslint-disable-next-line import/order
import lazyLoad from 'nuxt-lazy-load/lib/module'

const props = {
  bigCards: [
    {
      icon: {
        dimensions: {
          width: 45,
          height: 45,
        },
        alt: 'Gitlab Icon',
        copyright: null,
        url: 'https://superpupertest.cdn.prismic.io/superpupertest/4588b69e-4518-4ef0-b9eb-a17b793e06d3_gitlab_icon.svg',
      },
      isBig: true,
      title: 'Gitlab',
      description: 'A library for GPS and Accelerometer data "fusion" with Kalman filter.A library for GPS and Accelerometer data "fusion" with Kalman filter.',
    },
  ],

  littleCards: [
    {
      icon: {
        dimensions: {
          width: 45,
          height: 45,
        },
        alt: 'Gitlab Icon',
        copyright: null,
        url: 'https://superpupertest.cdn.prismic.io/superpupertest/4588b69e-4518-4ef0-b9eb-a17b793e06d3_gitlab_icon.svg',
      },
      isBig: false,
      title: 'Gitlab2',
      description: '',
    },
    {
      icon: {
        dimensions: {
          width: 45,
          height: 45,
        },
        alt: 'Gitlab Icon',
        copyright: null,
        url: 'https://superpupertest.cdn.prismic.io/superpupertest/4588b69e-4518-4ef0-b9eb-a17b793e06d3_gitlab_icon.svg',
      },
      isBig: false,
      title: 'Gitlab3',
      description: '',
    },
    {
      icon: {
        dimensions: {
          width: 45,
          height: 45,
        },
        alt: 'Gitlab Icon',
        copyright: null,
        url: 'https://superpupertest.cdn.prismic.io/superpupertest/4588b69e-4518-4ef0-b9eb-a17b793e06d3_gitlab_icon.svg',
      },
      isBig: false,
      title: 'Gitlab4',
      description: '',
    },
  ],
}

const localVue = createLocalVue()
localVue.directive('lazy-load', lazyLoad)
jest.mock('nuxt-lazy-load/lib/module')

describe('FourBigCards component', () => {
  it('should correctly render component', () => {
    const { container } = render(OneBigCards, {
      propsData: props,
      localVue,
    })

    const bigCards = screen.getAllByTestId('tech-big-card')
    const littleCards = screen.getAllByTestId('tech-little-card')

    expect(container).toMatchSnapshot()
    expect(bigCards).toHaveLength(props.bigCards.length)
    expect(littleCards).toHaveLength(props.littleCards.length)
  })

  it('should correctly return empty arrays from default props', () => {
    const wrapper = shallowMount(OneBigCards, {
      localVue,
    })

    expect(wrapper.vm.$options.props.bigCards.default.call()).toEqual([])
    expect(wrapper.vm.$options.props.littleCards.default.call()).toEqual([])
  })
})
