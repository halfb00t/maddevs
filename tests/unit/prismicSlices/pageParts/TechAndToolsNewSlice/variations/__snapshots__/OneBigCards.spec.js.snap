// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FourBigCards component should correctly render component 1`] = `
<div>
  <div
    class="one-big-cards"
  >
    <div
      class="technologies__big-card technologies__big-card--one-big-cards"
      data-testid="tech-big-card"
      isbig="true"
    >
      <div
        class="technologies__big-card-icon-wrapper"
      >
        <img
          alt="Gitlab Icon"
          class="technologies__big-card-icon"
          data-src="https://superpupertest.cdn.prismic.io/superpupertest/4588b69e-4518-4ef0-b9eb-a17b793e06d3_gitlab_icon.svg"
          height="45"
          width="45"
        />
      </div>
       
      <div
        class="technologies__big-card-info"
      >
        <h3
          class="technologies__big-card-title"
        >
          
      Gitlab
    
        </h3>
         
        <p
          class="technologies__big-card-description"
        >
          
      A library for GPS and Accelerometer data "fusion" with Kalman filter.A library for GPS and Accelerometer data "fusion" with Kalman filter.
    
        </p>
      </div>
    </div>
     
    <div
      class="technologies__little-card"
      data-testid="tech-little-card"
    >
      <img
        alt="Gitlab Icon"
        class="technologies__little-card-icon"
        data-src="https://superpupertest.cdn.prismic.io/superpupertest/4588b69e-4518-4ef0-b9eb-a17b793e06d3_gitlab_icon.svg"
        height="45"
        width="45"
      />
       
      <h3
        class="technologies__little-card-title"
      >
        
    Gitlab2
  
      </h3>
    </div>
    <div
      class="technologies__little-card"
      data-testid="tech-little-card"
    >
      <img
        alt="Gitlab Icon"
        class="technologies__little-card-icon"
        data-src="https://superpupertest.cdn.prismic.io/superpupertest/4588b69e-4518-4ef0-b9eb-a17b793e06d3_gitlab_icon.svg"
        height="45"
        width="45"
      />
       
      <h3
        class="technologies__little-card-title"
      >
        
    Gitlab3
  
      </h3>
    </div>
    <div
      class="technologies__little-card"
      data-testid="tech-little-card"
    >
      <img
        alt="Gitlab Icon"
        class="technologies__little-card-icon"
        data-src="https://superpupertest.cdn.prismic.io/superpupertest/4588b69e-4518-4ef0-b9eb-a17b793e06d3_gitlab_icon.svg"
        height="45"
        width="45"
      />
       
      <h3
        class="technologies__little-card-title"
      >
        
    Gitlab4
  
      </h3>
    </div>
  </div>
</div>
`;
